src/main/java/io/fabric8/maven/docker/AbstractDockerMojo.java

```diff
@@ -425,6 +430,19 @@ protected void checkImageWithAutoPull(ServiceHub hub, String image, String regis
         }
     }
 
+    private synchronized Set<String> getPreviouslyPulledImageCache() {
+        @SuppressWarnings({ "rawtypes", "unchecked" })
+        Set<String> cache = (Set) session.getUserProperties().get(CONTEXT_KEY_PREVIOUSLY_PULLED);
+        if (cache == null) {
+            System.out.println("creating new context");
+
+            cache = Sets.newConcurrentHashSet();
```

Comment:
i'm not sure what other sharing mechanisms we have and i'm going to guess that the plugin writes `user` properties and `system` properties right?

i don't want to make this overly complicated...what about a comma delimited list?

how do i trigger the integration tests locally?